frontend/app/controllers/spree/products_controller.rb

```diff
@@ -70,7 +70,7 @@ def load_variants
     end
 
     def load_products
-      retrieved = Rails.cache.fetch params.permit! do
+      retrieved = Rails.cache.fetch params.permit!.merge(current_currency: current_currency) do
```

Comment:
Should we update it here as well?
https://github.com/spark-solutions/spree/blob/db0f7a57c2d2fd4115dbc2d944ecae29c2d3d5c6/frontend/app/views/spree/products/index.html.erb#L39